op-node/service_l1_chain_config_test.go

```diff
@@ -0,0 +1,131 @@
+package opnode
+
+import (
+	"encoding/json"
+	"math/big"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+
+	nodeflags "github.com/ethereum-optimism/optimism/op-node/flags"
+)
+
+func TestNewL1ChainConfig_KnownChains(t *testing.T) {
+	logger := log.New()
+	app := cli.NewApp()
+	ctx := cli.NewContext(app, nil, nil)
+
+	t.Run("mainnet", func(t *testing.T) {
+		cfg, err := NewL1ChainConfig(new(big.Int).Set(params.MainnetChainConfig.ChainID), ctx, logger)
+		require.NoError(t, err)
+		require.Equal(t, params.MainnetChainConfig, cfg)
+	})
+
+	t.Run("sepolia", func(t *testing.T) {
+		cfg, err := NewL1ChainConfig(new(big.Int).Set(params.SepoliaChainConfig.ChainID), ctx, logger)
+		require.NoError(t, err)
+		require.Equal(t, params.SepoliaChainConfig, cfg)
+	})
+}
+
+func TestNewL1ChainConfig_CustomDirectAndEmbeddedAndNil(t *testing.T) {
+	logger := log.New()
+
+	// Build a minimal custom ChainConfig
+	custom := &params.ChainConfig{
+		ChainID:            big.NewInt(424242),
+		BlobScheduleConfig: &params.BlobScheduleConfig{},
+	}
+
+	customFaulty := &params.ChainConfig{
+		ChainID:            big.NewInt(424242),
+		BlobScheduleConfig: nil,
+	}
+
+	// Prepare temp dir
+	dir := t.TempDir()
+
+	// Direct JSON file containing a ChainConfig
+	directPath := filepath.Join(dir, "chainconfig.json")
+	{
+		f, err := os.Create(directPath)
+		require.NoError(t, err)
+		enc := json.NewEncoder(f)
+		err = enc.Encode(custom)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+	}
```

Comment:
I think this code could be deduplicated 3 times as a function of `path string, target any`, like

```suggestion
	encode := func (path string, target any) {
		f, err := os.Create(path)
		require.NoError(t, err)
		enc := json.NewEncoder(f)
		err = enc.Encode(target)
		require.NoError(t, err)
		require.NoError(t, f.Close())
	}
	encode(directPath, custom)
	encode(directFaultyPath, customFaulty)
	type wrapper struct {
		Config *params.ChainConfig `json:"config"`
	}
	encode(embeddedPath, wrapper{Config: custom})
```

Reply:
Good idea, f4027b0eacae7d4474f4d2aacd34c3b916b7276d